objectstore.go

```diff
@@ -0,0 +1,141 @@
+package civogo
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// ObjectStorage is the struct for the ObjectStorage model
+type ObjectStorage struct {
+	ID            string `json:"id"`
+	Name          string `json:"name"`
+	GeneratedName string `json:"generated_name"`
+	//default:1000
+	MaxObjects int `json:"max_objects"`
+	//default:500G
+	MaxSize         string `json:"max_size"`
+	AccessKeyID     string `json:"access_key_id"`
+	SecretAccessKey string `json:"secret_access_key"`
+	BucketURL       string `json:"bucket_URL"`
+	//Status can be one of - 1. Ready, 2.Creating and 3. failed
+	Status string `json:"status"`
+}
+
+// CreateObjectStorageRequest holds the request to create a new object storage
+type CreateObjectStorageRequest struct {
+	Name       string `json:"name" validate:"required"`
+	MaxSizeGB  int    `json:"max_size_gb" validate:"required"`
+	MaxObjects int    `json:"max_objects"`
+}
+
+// UpdateObjectStorageRequest holds the request to update a specified object storage's details
+type UpdateObjectStorageRequest struct {
+	MaxSizeGB  int `json:"max_size_gb"`
+	MaxObjects int `json:"max_objects"`
+}
+
+// ListObjectStores returns all objectstores in that specific region
+func (c *Client) ListObjectStores() ([]ObjectStorage, error) {
+	resp, err := c.SendGetRequest("/v2/objectstorage/buckets")
+	if err != nil {
+		return nil, decodeError(err)
+	}
+
+	var objectstores = make([]ObjectStorage, 0)
+	if err := json.NewDecoder(bytes.NewReader(resp)).Decode(&objectstores); err != nil {
+		return nil, err
+	}
+
+	return objectstores, nil
+}
+
+// GetObjectStore finds an objectstore by the full ID
+func (c *Client) GetObjectStore(id string) (*ObjectStorage, error) {
```

Comment:
Nice to see ObjectStore here, but the other methods are inconsistently named with ObjectStorage. They should be consistent.